### PR TITLE
Add support for symfony/validator

### PR DESCRIPTION
**Pull request by enumag**
Needed for https://github.com/Kdyby/Validator/pull/16.

I don't like this solution though. Let me think about it for a bit. Suggestions are welcomed of course.

Also there is obviously missing support for traits.

### CODE REVIEW

File: src/Kdyby/DoctrineCache/Cache.php [hunk 1]
```diff
@@ -96,12 +96,13 @@ protected function doSave($id, $data, $lifeTime = 0)
 				$files[] = ClassType::from($class)->getFileName();
 			}
 		}
+		if ($data instanceof \Symfony\Component\Validator\Mapping\ClassMetadata) {
+			$files[] = ClassType::from($data->name)->getFileName();
+		}
 
 		if (!empty($data)){
-			if ((is_array($data) && reset($data) instanceof Doctrine\ORM\Mapping\Annotation) || $data instanceof Doctrine\ORM\Mapping\Annotation) {
```
Comment (fprochazka) [inline src/Kdyby/DoctrineCache/Cache.php]: I'm not happy about this change, why have you modified it?

Comment (enumag) [inline src/Kdyby/DoctrineCache/Cache.php]: Because validator constraints are obiously not instances of `Doctrine\ORM\Mapping\Annotation`? :-)

Comment (fprochazka) [inline src/Kdyby/DoctrineCache/Cache.php]: And what about doctrine? :)

Comment (enumag) [inline src/Kdyby/DoctrineCache/Cache.php]: I don't understand, I removed the condition so now it will accept any class. Is that a problem?

Comment (fprochazka) [inline src/Kdyby/DoctrineCache/Cache.php]: I see now, I should get more sleep :)

Is there anything apart from annotations, that is using the same cache id scheme?
I guess it's not a problem to check every key, but it seemed unnecesary.
I guess we can keep this then :)

**Comment by fprochazka** [comment]
I think it's fine. Working is better then perfect :) We can make it nicer when we figure out how.

**Comment by enumag** [comment]
In my opinion there should be `DependenciesProviderInterface` which would receive the data and return dependencies for nette cache. Implementations of this interface should be in kdyby/doctrine and kdyby/validator respectively.

But I wan't to check how Symfony is handling this or whether it has such invalidation at all (couldn't find it in the code anywhere so I need to install it and try).

File: src/Kdyby/DoctrineCache/Cache.php (hunk 1 above)
Comment (enumag) [inline src/Kdyby/DoctrineCache/Cache.php]: > Is there anything apart from annotations, that is using the same cache id scheme?

Not that I know of but that doesn't mean anything...

My proposal about `DependenciesProviderInterface` would solve that though.

**Comment by fprochazka** [comment]
@enumag that sounds like a clean solution, but unnecesary slow one. Cache should be fast.

**Comment by enumag** [comment]
@fprochazka It's only for debug mode so it doesn't matter that much right?

**Comment by fprochazka** [comment]
@enumag good point

**Comment by enumag** [comment]
@fprochazka Should I implement it then?

**Comment by fprochazka** [comment]
@enumag sure, if you have the time :)

**Comment by enumag** [comment]
@fprochazka Well not right now, but I'll try to make some next week. :-)

**Pull request merged**